common/event/emitter.js

```diff
@@ -1,51 +1,71 @@
 /**
```

Comment:
Still skeptical with this emitter's implementation - it lacks once and message passing / stream support.  If the intend is for really basic emitter usage, then i would be okay with it.